.codesandbox/Dockerfile

```diff
@@ -0,0 +1,11 @@
+# Use the node:20-alpine base image
+FROM node:20-alpine
+
+# Install zsh and Git
+RUN apk update && apk add --no-cache zsh git
```

Comment:
this needs to be documented, maybe with a link to why this is

> Without zsh, the dependency installation is throwing an error, so I added it
> ```
> OCI runtime exec failed: exec failed: unable to start container process: exec: "/bin/zsh": stat /bin/zsh: no such file or directory: unknown
> ```
> Without git, the terminal after forking is not working

we could also consider using `slim` instead, since CodeSandbox uses those in [one of their test `Dockerfile`s](https://github.com/codesandbox/codesandbox-client/blob/aed183fa49072c18adf3d75671ac9116e3b69e4f/docker/Dockerfile.test#L4)

Reply:
Switched to `slim` instead and in this step, I had to switch to `apt-get` because this is the package manager included in `node:20-slim`. Additionally removed the `--repository` option,  which is specific to `apk`.

[876a08a](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/876a08a1b9efa4dbe18556b919a124bec033949d)

Still working on the comments and the links that I want to include

Reply:
Changed to slim. With `slim` I need to change `apk` to `apt-get`, because `apt-get` is the default package manager in `node:20-slim` 
[876a08a](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/876a08a1b9efa4dbe18556b919a124bec033949d)

Reply:
thinking about it more, it would be nice to be able to document what additional packages CodeSandbox needs to have installed for `node:20-alpine`

Reply:
I am testing what additional software we need for CodeSandbox to run with `node:20-alpine`. With the `alpine` version I can use `apk` to add the software. Without adding any, and only using `node:20-alpine`, the first error message occurs when CodeSandbox installs the dependencies. 
```
OCI runtime exec failed: exec failed: unable to start container process exec: "/bin/zsh": no such file or directory: unknown
```
<img width="607" alt="Screenshot 2024-05-15 at 14 46 52" src="https://github.com/upleveled/express-guest-list-api-memory-data-store/assets/80746311/37fe6a1e-186e-4931-a39b-effa734f05a1">

This error tells me that CodeSandbox needs `zsh` to install the dependencies. After adding `zsh`, all dependencies get installed. After opening the terminal, the next error message occurs
```
command not found: git
```
<img width="619" alt="Screenshot 2024-05-15 at 14 49 01" src="https://github.com/upleveled/express-guest-list-api-memory-data-store/assets/80746311/7e3d6701-d70d-4a3e-abd4-3fc826bc3184">

After adding `git` to use the terminal, the project runs as intended.

Interesting at the moment is that I don't have to add `yq`, to add the pnpm version, that we are using inside our `package.json` file. Normally `yq` is not pre-installed and we have to install it, but the correct version I declared in my `package.json` is active on CodeSandbox.

<img width="619" alt="Screenshot 2024-05-15 at 14 50 03" src="https://github.com/upleveled/express-guest-list-api-memory-data-store/assets/80746311/27301d1c-dfd4-4499-a41c-bcb7554761b1">

Commit [44212ae](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/44212aead4f2e6c805e46b56da0334495ec88503)

Reply:
one other idea I had just now: maybe we can configure the default shell in CodeSandbox to use `bash` (which I believe is preinstalled on `node:20-alpine`) instead of using `zsh`... 🤔

Reply:
oh - I looked a bit through the docs and confirmed with Ives - there is no such setting yet

but CodeSandbox may build this feature in future (setting the default shell) - it should be possible when full VS Code is available

so let's add a TODO in the Dockerfile to configure the shell to `bash` in future when/if this feature appears in CodeSandbox

Reply:
Added a TODO comment in the Dockerfile 
[6a4c08f](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/6a4c08f07ef2c371441ce4b32ff0b0e4944e9634)

Reply:
we lost the docs that you had there before documenting why the 3 packages need to be installed

Reply:
```dockerfile
FROM node:lts-alpine

# 1. zsh is default shell on CodeSandbox - without it,
# the devbox fails to start with a "OCI runtime exec
# failed" error:
# ```
# OCI runtime exec failed: exec failed: unable to start container process: exec: '/bin/zsh': stat /bin/zsh: no such file or directory: unknown
# ```
#
# 2. Git is a dependency of zsh - without it, opening
# the terminal fails with a "command not found" error:
# ```
# "/root/.oh-my-zsh/tools/check_for_upgrade.sh:31: command not found: git"
# ```
#
# 3. yq used to install matching version of pnpm
#
# TODO: Switch to bash and remove the zsh packages in
# future if CodeSandbox adds a feature to configure
# the shell
# https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294#discussion_r1601759357
RUN apk update && apk add --no-cache zsh git yq

RUN corepack enable && corepack prepare pnpm@$(yq e '.engines.pnpm' package.json) --activate
```

Reply:
Updated the comment in this commit [ab31bde](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/ab31bde7ea8b3d210d78d043df16525d4714683c)

Added the TODO comment and used rewrap in this commit
[fab36de](https://github.com/upleveled/express-guest-list-api-memory-data-store/pull/294/commits/fab36de218a9dd7808be640d35ff007dfc318291)